File: app/src/main/java/ru/application/homemedkit/connectionController/models/DrugsData.java
package ru.application.homemedkit.connectionController.models;

import static ru.application.homemedkit.helpers.ConstantsHelper.BLANK;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DrugsData {
    @JsonProperty("prodDescLabel")
    @JsonSetter(nulls = Nulls.SKIP)
    public String prodDescLabel = BLANK;
    @JsonProperty("foiv")
    @JsonSetter(nulls = Nulls.SKIP)
    public Foiv foiv = new Foiv();
    @JsonProperty("expireDate")
    @JsonSetter(nulls = Nulls.SKIP)
    public long expireDate = -1L;
    @JsonProperty("vidalData")
    @JsonSetter(nulls = Nulls.SKIP)
    public VidalData vidalData = new VidalData();
}

File: app/src/main/java/ru/application/homemedkit/connectionController/models/DrugsData.kt
package ru.application.homemedkit.connectionController.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.application.homemedkit.helpers.BLANK

@JsonClass(generateAdapter = true)
data class DrugsData(
    @Json val prodDescLabel: String = BLANK,
    @Json val foiv: Foiv = Foiv(),
    @Json val expireDate: Long = -1L,
    @Json val vidalData: VidalData = VidalData()
)
